ntoskrnl/mm/ARM3/procsup.c

```diff
@@ -97,6 +97,13 @@ MiCreatePebOrTeb(IN PEPROCESS Process,
         return STATUS_NO_MEMORY;
     }
 
+    Status = PsChargeCurrentProcessNonPagedPoolQuota(sizeof(MMVAD_LONG));
+    if (!NT_SUCCESS(Status))
+    {
+        ExFreePoolWithTag(Vad, 'ldaV');
+        return STATUS_QUOTA_EXCEEDED;
```

Comment:
`PsChargeProcessNonPagedPoolQuota` already returns `STATUS_QUOTA_EXCEEDED` for you in case quota limits have been reached. Normally what you'd need to do when a function that returns a NTSTATUS fails is just that, e.g -
```
if (!NT_SUCCESS(Status))
{
    /* Do whatever.... */
    [...]
    return Status;
}
```

Reply:
Yeah you're right, in this case I figured it'd be good for readability. But no, you're right. Will fix! ty!